warehouse/query-core/src/main/java/datawave/query/tables/RangeStreamScanner.java

```diff
@@ -135,28 +153,212 @@ public RangeStreamScanner setScannerFactory(ScannerFactory factory) {
     
     /**
      * Override this for your specific implementation.
-     * 
+     *
      * In this specific implementation our row key will be the term, the column family will be the field name, and the column family will be the shard,so we
      * should have the following as our last key
-     * 
+     *
      * bar FOO:20130101_0
-     * 
+     *
      * so we should append a null so that we we don't skip shards. similarly, an assumption is made of the key structure within this class.
-     * 
+     *
      * @param lastKey
      * @param previousRange
      */
     @Override
     public Range buildNextRange(final Key lastKey, final Range previousRange) {
-        
         /*
          * This path includes the following key from the shard_id onward. The reason we also append the hex 255 value is because we receive a key not unlike
          * foo:20130101_0. If our next search space is foo:20130101_0\x00 we will hit all data types within that range...again..and again...and again. To
          * account for this, we put \uffff after the null byte so that we start key is technically the last value within the provided shard, moving us to the
          * exact next key within our RangeStream
          */
-        return new Range(new Key(lastKey.getRow(), lastKey.getColumnFamily(), new Text(lastKey.getColumnQualifier() + "\uffff")), true,
-                        previousRange.getEndKey(), previousRange.isEndKeyInclusive());
+        Key startKey = new Key(lastKey.getRow(), lastKey.getColumnFamily(), new Text(lastKey.getColumnQualifier() + "\uffff"));
+        return new Range(startKey, true, previousRange.getEndKey(), previousRange.isEndKeyInclusive());
+    }
+    
+    /**
+     * Seek range is built with a start key column qualifier set to the seek shard.
+     *
+     * In a worst case scenario the built seek range is for the very next shard, no differently than {@link #buildNextRange}.
+     *
+     * Best case the seek range lies beyond the data available for this scanner, thus negating any need to continue searching within this range stream.
+     *
+     * @param seekShard
+     *            the shard to seek to
+     * @param range
+     *            the RangeStreamScanner's current range
+     * @return the current range with a modified start key
+     */
+    public Range buildSeekRange(String seekShard, Range range) {
+        
+        // Adjust startKey columnQualifier to be the shard.
+        Key startKey = range.getStartKey();
+        Text row = startKey.getRow();
+        Text cf = startKey.getColumnFamily();
+        startKey = new Key(row, cf, new Text(seekShard));
+        
+        return new Range(startKey, true, range.getEndKey(), false);
```

Comment:
isEndKeyInclusive() or false?

Reply:
Should be isEndKeyInclusive, good catch.